pkg/manifests/tls_test.go

```diff
@@ -88,23 +93,74 @@ func TestNeedsNewCert(t *testing.T) {
 func TestRotateGrpcTLSSecret(t *testing.T) {
 	f := NewFactory("openshift-monitoring", "openshift-user-workload-monitoring", NewDefaultConfig())
 
-	s, err := f.GRPCSecret(nil)
-	if err != nil {
-		t.Fatal(err)
-	}
+	for _, tc := range []struct {
+		name  string
+		setup func(*v1.Secret)
+		test  func(spre, spost *v1.Secret) error
+	}{
+		{
+			name: "no rotation without modification",
+			test: func(spre, spost *v1.Secret) error {
+				if !reflect.DeepEqual(spre.Data, spost.Data) {
+					return fmt.Errorf("expected certificate data to be equal, but it isn't, pre-rotation key material %v, post-rotation key material %v", spre, spost)
+				}
+				return nil
+			},
+		},
+		{
+			name: "force rotation",
+			setup: func(s *v1.Secret) {
+				s.Annotations["monitoring.openshift.io/grpc-tls-forced-rotate"] = "true"
```

Comment:
Could we add an Annotation with`blah/foo-bar`? Would it make sense to have that testcase?

Reply:
The expected behavior would be a no-op, not sure about the benefit, but I can add a test case